### ISSUE TITLE
GML driver silently ignores fields with type OFTTime

### ISSUE THREAD

**Issue by rbuffat** [OSGeo/gdal#2482]
I'm not sure if this is a bug or is intended behavior. The GML driver silently ignores fields with type ogr.OFTTime. I'm not familiar with the GML format specification, but after briefly skimming the specification I assume that time is not a supported field type. Thus, I was wondering if it would be possible to issue a warning / error when a field of this type is created. 

Sample Code:
```
from osgeo import gdal
from osgeo import ogr
import sys

print("GDAL:", gdal.__version__)

gdal.UseExceptions()

driverName = 'GML'


def error_handler(err_level, err_no, err_msg):
    print(err_level, err_no, err_msg)


gdal.PushErrorHandler(error_handler)


for field_type, filename in [(ogr.OFTDateTime, 'test_datetime.gml'),
                             (ogr.OFTDate, 'test_date.gml'),
                             (ogr.OFTTime, 'test_time.gml')]:

    drv = gdal.GetDriverByName(driverName)

    ds = drv.Create(filename, 0, 0, 0, gdal.GDT_Unknown)
    if ds is None:
        print("Creation of output file failed.\n")
        sys.exit(1)

    lyr = ds.CreateLayer("abc", None, ogr.wkbMultiPolygon25D)
    if lyr is None:
        print("Layer creation failed.\n")
        sys.exit(1)

    field_defn = ogr.FieldDefn("datetimefield", field_type)
    if lyr.CreateField(field_defn) != 0:
        print("Creating field failed.\n")
        sys.exit(1)

    point = ogr.Geometry(ogr.wkbPoint)
    point.AddPoint(1198054.34, 648493.09)

    feat = ogr.Feature(lyr.GetLayerDefn())
    feat.SetField("datetimefield", 2020, 1, 2, 3, 4, 5.0, 0)
    feat.SetGeometry(point)

    if lyr.CreateFeature(feat) != 0:
        print("Failed to create feature in shapefile.\n")
        sys.exit(1)

    feat.Destroy()

    ds = None

for field_type, filename in [(ogr.OFTDateTime, 'test_datetime.gml'),
                             (ogr.OFTDate, 'test_date.gml'),
                             (ogr.OFTTime, 'test_time.gml')]:
    print("")
    print("Driver: {}".format(driverName))
    print("File: {}".format(filename))

    driver = ogr.GetDriverByName(driverName)
    dataSource = driver.Open(filename, 0)
    layer = dataSource.GetLayer()
    for feat in layer:
        print("Number of fields {}".format(feat.GetFieldCount()))
        print(feat.GetFieldAsString("datetimefield"))

```

Output:
```
GDAL: 3.0.4

Driver: GML
File: test_datetime.gml
Number of fields 2
2020/01/02 03:04:05

Driver: GML
File: test_date.gml
Number of fields 2
2020/01/02

Driver: GML
File: test_time.gml
Number of fields 1
3 1 Invalid field name: 'datetimefield'
None
```

**Comment by rouault** [comment]
Was fixed in upcoming 3.1 per 8f2d9a25ddfba66412b29ced2eda889ef6bea5dd

**Comment by rbuffat** [comment]
@rouault Thanks! Sorry for causing extra work, I didn't spot this.
